File: Interviews/LeetCode/Topic-Wise/STL/Medium/Continuous-Subarray-Sum/523.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> ump;
        int sum = 0, n = nums.size();
        ump[0] = -1;
        for (int i = 0; i < n; i++) {
            sum = (sum + nums[i]) % k;
            if (ump.find(sum) != ump.end()) {
                if (i - ump[sum] > 1) return true;
            } else {
                ump[sum] = i;
            }
        }

        return false;
    }
};

File: Interviews/LeetCode/Topic-Wise/STL/Medium/Continuous-Subarray-Sum/523.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        ump, curSum, n = {}, 0, len(nums)
        ump[0] = -1
        for i in range(n):
            curSum = (curSum + nums[i]) % k
            if curSum in ump:
                if i - ump[curSum] > 1: return True
            else:
                ump[curSum] = i
        
        return False